### PR TITLE
Nuxt removal

### LINKED ISSUE #5968: Remove Nuxt as a Dependency
Since we're not using Nuxt for very much functionality we're looking to remove. Ultimately we're trying to simplify the project and make the code more explicit.

TODO:

- Verify that building and building with plugins is still functional. Thus far I've only verified that the local version of nuxt removal works. We still need to make sure that normal and plugin release versions still work. (was tracked by rancher/dashboard#6439)
- Live/hot module reload aren't working in the nuxt removal branch. I think this will be a relatively simple webpack configuration task. I think it's just missing rather than being broken. (was tracked by rancher/dashboard#6437)

### PR DESCRIPTION
**Pull request by codyrancher**
# General strategy:
Nuxt generates files for our application including things like route creation or plugin instantiation. I take all of the generated files and move them into shell/nuxt with minimal changes. I then migrate our nuxt.config.js to vue.config.js as close to 1 to 1 as I can. I then modify certain files if we no longer support certain aspects such as not supporting .ts/module imports in the vue.config.js or having to prepend @ to some of the style file imports. 

# Notes
- I don't think you need to look at the files in the nuxt directory. If you insist, the two that have the most edits are the router.js and the index.js files in order to fix imports.
- Changes like https://github.com/rancher/ui-plugin-examples/pull/17/files will be necessary for extensions to work with this change 
- I verified that extensions work with the ui-plugin-examples
- I still need to verify that the dist directory generated by yarn build will work

fixes rancher/dashboard#5968

### CODE REVIEW

File: .github/workflows/test.yaml
Comment (codyrancher) [inline .github/workflows/test.yaml:48]: The e2e tests were failing due to coverage failing.
Coverage was failing complaining that out.json didn't exist so I created it. The tests started passing but I'm not certain the coverage was actually getting generated.

**Review by richard-cox — CHANGES_REQUESTED** [review]
I given this a quick run though, one comment, everything else inline

- I couldn't access the dashboard once up and running
  - Tried both `API=... yarn mem-dev` and `API=... yarn dev`
  - Reported `https://<hostname>:8005/` gave dns error. `https://localhost:8005/` was not reachable

File: .github/workflows/test.yaml
Comment (cnotv) [reply .github/workflows/test.yaml:48]: This should be automatically generated and if missing there must be some configuration issues or mismatching versions.
On a first glance, this is related to a lack of instrumentalization of the tests required to generate the coverage afterwards.
This has been added as [babel handler within Nuxt plugins as `'babel-plugin-istanbul'`](https://github.com/rancher/dashboard/blob/master/shell/nuxt.config.js#L545).

File: cypress.config.ts
Comment (cnotv) [reply cypress.config.ts:33]: @codyrancher please keep the coverage false or it will be generated every time we run the test. The CI has already a flag to enable it.

**Review by cnotv — CHANGES_REQUESTED** [review]

File: shell/package.json
Comment (codyrancher) [reply shell/package.json:19]: In nuxt I had to use mem-dev otherwise I couldn't build our dashboard. I no longer needed mem-dev to so I removed this. If we want to add it back we can but it seems like we should get rid of things that are no longer necessary.

File: shell/assets/styles/fonts/_fontstack.scss
Comment (codyrancher) [reply shell/assets/styles/fonts/_fontstack.scss:7]: fixed

File: shell/utils/gc/gc-types.ts
Comment (codyrancher) [reply shell/utils/gc/gc-types.ts:1]: This was another case of us importing into vue.config.js. I'd like to eventually get imports and ts files to work with vue.config but I don't know if it's even possible right now.

File: .github/workflows/test.yaml
Comment (cnotv) [reply .github/workflows/test.yaml:48]: Please do not forget to remove the change to the test, as it already should highlight an issue in a configuration of the project, rather than an issue in the configuration of the CI.

File: scripts/build-embedded
Comment (codyrancher) [reply scripts/build-embedded:24]: Fixed

**Comment by codyrancher** [comment]
> I given this a quick run though, one comment, everything else inline
> 
>     * I couldn't access the dashboard once up and running
>       
>       * Tried both `API=... yarn mem-dev` and `API=... yarn dev`
>       * Reported `https://<hostname>:8005/` gave dns error. `https://localhost:8005/` was not reachable

We'll need to take a look together to figure out the issue. I pulled a fresh copy and I can still run and see the dashboard without issue. There's also obviously nothing related to DNS in these changes.

**Comment by codyrancher** [comment]
@cnotv resolved both of the issues you mentioned. Looks like it was just the missing istanbul plugin that was causing the issues I ran into.

https://github.com/rancher/dashboard/pull/7211#discussion_r1033356172
https://github.com/rancher/dashboard/pull/7211#discussion_r1033356507

File: shell/vue.config.js
Comment (codyrancher) [reply shell/vue.config.js:1]: I did another sync. There are certain things omitted purposefully. If you see anything you think I missed or want clarification why I omitted maybe we can talk about those specifics.

File: shell/nuxt/middleware.js
Comment (codyrancher) [reply shell/nuxt/middleware.js:1]: Added docs for routes, middleware, plugins and stores.

File: pkg/harvester/components/HarvesterUpgradeHeader.vue
Comment (richard-cox) [inline pkg/harvester/components/HarvesterUpgradeHeader.vue:5]: Preferably all these changes from these should stay as `@/pkg`, I think that might break things somewhere. The `check-plugin` github workflow will fail if there's any in i think shell or a test pkg.

Comment (richard-cox) [reply pkg/harvester/components/HarvesterUpgradeHeader.vue:5]: As a fall back, they could be made relative rather than absolute

File: shell/public/index.html
Comment (richard-cox) [inline shell/public/index.html:5]: In the old world build in packages (pkgs in `/pkg` that were `autoLoad`ed via `autoLoadPackages` in nuxt.config were brought in here. `autoLoadPackages` does exist in vue.config.js but isn't used anywhere. Does this feature still work?

File: shell/vue.config.js
Comment (richard-cox) [reply shell/vue.config.js:1]: There's a few things that have crept in, `skipEsLintCheck` and `path.posix.join` (the former helps a lot when building shell when its a dependency

File: shell/utils/gc/gc-types.ts
Comment (richard-cox) [reply shell/utils/gc/gc-types.ts:1]: It should be possible to convert vue.config to typescript, there's a couple of methods in the main github issue for it
- https://github.com/vuejs/vue-cli/issues/2138#issuecomment-581125115 / https://github.com/vuejs/vue-cli/issues/2138#issuecomment-1010837395
- https://github.com/vuejs/vue-cli/issues/2138#issuecomment-677635585 (bit yanky though)
- https://github.com/vuejs/vue-cli/issues/2138#issuecomment-1152832773

File: scripts/build-embedded
Comment (richard-cox) [reply scripts/build-embedded:24]: For reference this is the PR https://github.com/rancher/dashboard/pull/7516

File: shell/config/private-label.js
Comment (richard-cox) [reply shell/config/private-label.js:1]: Hopefully if we can convert vue.config to ts we shouldn't need to make these changes

**Review by richard-cox — CHANGES_REQUESTED** [review]
At some point i'll set up a call and we can debug the issue i have reaching `localhost:8005`

- `yarn build-pkg harvester` output to `dist`, rather than `dist-pkg`
- the yarn `app` creator will need updating, it contains references to things like `nuxt.config.js`.
- i couldn't see where the functionality from `nuxt.config.js` was applied in the new world, does that get sucked in as `_appConfig` in vue.config? If so where's the default file gone (there's probably docs somewhere that reference it)?

**Comment by cnotv** [comment]
@richard-cox could you please verify the issue about the branding E2E error failing, since you are the actual author?

File: pkg/harvester/components/HarvesterUpgradeHeader.vue
Comment (cnotv) [reply pkg/harvester/components/HarvesterUpgradeHeader.vue:5]: I think this is related to the configuration we have in this project which is messing up also with the intellisense. Every time there's an automatic import, it changes to it. Should we create a further issue or is sufficient the [existing one](https://github.com/rancher/dashboard/issues/7357) (maybe with higher weight)?

**Comment by codyrancher** [comment]
It turns out the branding tests were failing due to errors in the labeled-input and checkbox-input po objects where there was an extraneous space in the invocation of the contains method.

@richard-cox @cnotv I'd appreciate it if we could get a ship it on everything but the vue.config.js. There's even more changes there since you last reviewed and I'd like to avoid merging this file over and over.

**Comment by cnotv** [comment]
> It turns out the branding tests were failing due to errors in the labeled-input and checkbox-input po objects where there was an extraneous space in the invocation of the contains method.
> 
> @richard-cox @cnotv I'd appreciate it if we could get a ship it on everything but the vue.config.js. There's even more changes there since you last reviewed and I'd like to avoid merging this file over and over.

@richard-cox

File: shell/utils/gc/gc-types.ts
Comment (codyrancher) [reply shell/utils/gc/gc-types.ts:1]: Looks like I was able to get this to work. I'm going to make it a separate commit so it can be easily removed in case it causes issues.

File: shell/config/private-label.js
Comment (codyrancher) [reply shell/config/private-label.js:1]: done as mentioned above

File: docusaurus/docs/code-base-works/plugins.md
Comment (codyrancher) [reply docusaurus/docs/code-base-works/plugins.md:1]: Done

File: docusaurus/docs/code-base-works/routes.md
Comment (codyrancher) [reply docusaurus/docs/code-base-works/routes.md:5]: Done

File: shell/config/private-label.js
Comment (richard-cox) [reply shell/config/private-label.js:1]: The vue.config file is still `js`. We could avoid losing TS files though by moving the definition of `STANDARD` (with any  and custom) out in to their won file

File: shell/utils/gc/gc-types.ts
Comment (richard-cox) [reply shell/utils/gc/gc-types.ts:1]: Given the comment for private-label.js we should be able to keep these as TS without converting vue.config to ts

**Review by richard-cox — CHANGES_REQUESTED** [review]
Aside from a bump to vue.config for changes in nuxt.config there's a number of unresolved comments (all in PR review history)
- testing https://github.com/rancher/dashboard/pull/7516 after a bump
- skipLibCheck question
- keeping some files as TS (even with vue.config as js)
- updates to yarn app creator
- updates made to root `nuxt.config.js` are missing
- harvester import updates to ~/pkg (might need something similar to https://github.com/rancher/dashboard/pull/7676)
- `autoLoadPackages` isn't wired in (build in `/pkg` are skipped

In addition
- `yarn build-pkg harvester` fails
- I gave the dashboard a small click around and discovered
  - There's errors in the console
    ```
    https://linux-o35c:8005/sockjs-node/info?t=1671461124843 net::ERR_CERT_AUTHORITY_INVALID
    ``` 
  - After installing the Extensions operator and the Elemental plugin, refreshing whilst on the Elemental dashboard results in a fail-whale page `Type dashboard not found, unable to display list`. I couldn't see this on a deployed system. Might need to check with @aalves08 
    ```
    TypeError: Cannot read properties of undefined (reading 'id')
        at Object.eval (type-map.js?9bc6:929:1)
    ```
On the plus side I can now run and access the dashboard. Adding an entry `hosts` entry for my machine name works, or supplying `--host localhost` when running `yarn dev` both worked. We'll need to monitor if this fails for other people, but at the moment I don't think anything is required

File: pkg/harvester/components/HarvesterUpgradeHeader.vue
Comment (codyrancher) [reply pkg/harvester/components/HarvesterUpgradeHeader.vue:5]: I was able to fix this at least at build time with a new alias. It doesn't look like it broke anything.

File: shell/public/index.html
Comment (codyrancher) [reply shell/public/index.html:5]: What gives you the impression autoLoadPackages was used here before? Can you link to an example because 
it looks like it's not used anywhere https://github.com/rancher/dashboard/search?q=autoLoadPackages&type=

This was just copied over as part of the initial conversion.

File: shell/config/private-label.js
Comment (codyrancher) [reply shell/config/private-label.js]: It was already done just not committed. I like commenting as if it's a checklist.

File: shell/utils/gc/gc-types.ts
Comment (codyrancher) [reply shell/utils/gc/gc-types.ts]: Fixed when converting to ts.

File: scripts/build-embedded
Comment (codyrancher) [reply scripts/build-embedded:24]: This is the part that adds the version env variable https://github.com/rancher/dashboard/blob/92a6911b1aa3c7b6b5d031bb8f87649d3af89c1f/scripts/build-embedded#L16

I did have to make changes to the vue.config.js since it doesn't have a concept of publicRuntimeConfig. I defined that using definePlugin and updated the nuxt/client.js since definePlugin didn't seem to support replacing the NUXT prefix.

**Comment by codyrancher** [comment]
- testing https://github.com/rancher/dashboard/pull/7516 after a bump
  - I verified that this is working with my changes 
- skipLibCheck question
  -  I think you mean skipEsLint? If so, eslint doesn't run as a part of the vue watcher by default https://eslint.vuejs.org/user-guide/ I added it as a part of the build commands to still catch problems
- keeping some files as TS (even with vue.config as js)
  - This is done. It just wasn't pushed before you commented again. It's there now. 
- updates to yarn app creator
  - I'm not really sure what the ask is here 
- updates made to root nuxt.config.js are missing
  - Updated 
- harvester import updates to ~/pkg (might need something similar to https://github.com/rancher/dashboard/pull/7676)
  - Fixed by going to what's currently in master and updating the root vue.config 
- autoLoadPackages isn't wired in (build in /pkg are skipped
  - Commented above but I don't think this is actually getting used in master. Can you link if you know where it's used?  

I've updated the shell/vue.config.js as well. Here are some notes regarding things I didn't update:
- The plugins are imported in shell/index.js so I don't think there's a need for NUXT_SHELL.
- path.posix.join was specific to configuring nuxt directories. These won't affect the already generated code and aren't needed here
- As mentioned above eslint is not run as a part of the vue watcher so that can be skipped
- typescript field is omitted because it's specific to eslint which isn't a part of this config

File: shell/public/index.html
Comment (richard-cox) [reply shell/public/index.html:5]: Update to follow in main review comment, but this is applied with https://github.com/rancher/dashboard/blob/master/shell/nuxt.config.js#L588

**Review by richard-cox — CHANGES_REQUESTED** [review]
> testing https://github.com/rancher/dashboard/pull/7516 after a bump
> - I verified that this is working with my changes

I didn't see this working on the latest build

image

vs

image

> skipLibCheck question

I couldn't find references to skipEsLintCheck, skipLibCheck or skipEsLint. Not sure why I raised it, but can now be ignored.

> keeping some files as TS (even with vue.config as js)

Just the one comment about settings functions

> updates to yarn app creator

Discussed in slack

> updates made to root nuxt.config.js are missing

Looks good! I'll check again just before this merges

> autoLoadPackages isn't wired in (build in /pkg are skipped
> - Commented above but I don't think this is actually getting used in master. Can you link if you know where it's used?

I got confused on what the `autoLoad` does. In vue.config.ts
- `autoLoad` will stick the referenced package from ./pkg/<ref>/<ref>.umd.min.js (unless in the `excludes` value) into the DOM and then `$plugin.loadAsync` from `shell/plugins/plugin.js`
  - Note - this is wired in via https://github.com/rancher/dashboard/blob/master/shell/nuxt.config.js#L588 , which does not happen in the new world
- further down packages in `pkg` directories are brought in via another method.

@nwmac How do these two features work together, is `autoLoad` still needed? Can it be removed or should it be fixed in the new vue world?

> I gave the dashboard a small click around and discovered. There's errors in the console. https://linux-o35c:8005/sockjs-node/info?t=1671461124843 net::ERR_CERT_AUTHORITY_INVALID.
> - I haven't seen this. It sounds like you're having a cert issue with that host? Any more info for repro?

I double checked and I don't see this running `master` against the same Rancher instance and there's no requests to anything on that path. There's a few other people with the same issue (https://stackoverflow.com/questions/55754906/errors-in-browser-console-requests-to-sockjs-node-infot-1555629946494), that led me to adding 
```
 public: `https://0.0.0.0:${ devPorts ? 8005 : 80 }`,
```
to `shell/vue.config.ts` config.devServer which solved the problem. Can this be added? The change also resolved the issue i was seeing with hot-reload (interestingly it looks like that's done over websocket now, not sure how it worked before).

**Comment by codyrancher** [comment]
>> testing https://github.com/rancher/dashboard/pull/7516 after a bump
>> - I verified that this is working with my changes
> I didn't see this working on the latest build
> image
> vs
> image
> Update: This is probably caused by the `source` change to the `dev` targets in package.json

TMI But to explain this one... This was caused because the dev script omitted the `source ./scripts/version` prefix. I had verified and tested by setting `DASHBOARD_VERSION` env var which did work. I also found out that yarn uses it's own shell which you can set using `yarn config set script-shell /bin/bash` and I found that out because the default shell didn't have access to `source`.

> > keeping some files as TS (even with vue.config as js)
> Just the one comment about settings functions

Forgot that file. Now fixed.

> > updates to yarn app creator
> Discussed in slack

This was a massive pain to figure out and took multiple days and was ultimately a small change. When I first posted this PR this change would've been simple as I had also converted the ui-plugin-examples https://github.com/rancher/ui-plugin-examples/pull/17/files. I really think we need to stop cycling on this PR. Many of the issues raised in this PR could've been resolved in subsequent changes and I think very few would've actually been blockers. At least one of the requests was to account for a change that hadn't even been pushed yet (the versioning).

> > updates made to root nuxt.config.js are missing
> Looks good! I'll check again just before this merges

I think I'll also have to make a pass at the routes and plugins again before we merge.

> > autoLoadPackages isn't wired in (build in /pkg are skipped
> > - Commented above but I don't think this is actually getting used in master. Can you link if you know where it's used?
> 
> I got confused on what the `autoLoad` does. In vue.config.ts
> - `autoLoad` will stick the referenced package from ./pkg/<ref>/<ref>.umd.min.js (unless in the `excludes` value) into the DOM and then `$plugin.loadAsync` from `shell/plugins/plugin.js`
>   - Note - this is wired in via https://github.com/rancher/dashboard/blob/master/shell/nuxt.config.js#L588 , which does not happen in the new world
> - further down packages in `pkg` directories are brought in via another method.
> 
> @nwmac How do these two features work together, is `autoLoad` still needed? Can it be removed or should it be fixed in the new vue world?

@nwmac is there anything that needs to be done here?

> > I gave the dashboard a small click around and discovered. There's errors in the console. https://linux-o35c:8005/sockjs-node/info?t=1671461124843 net::ERR_CERT_AUTHORITY_INVALID.
> > - I haven't seen this. It sounds like you're having a cert issue with that host? Any more info for repro?
> I double checked and I don't see this running `master` against the same Rancher instance and there's no requests to anything on that path. There's a few other people with the same issue (https://stackoverflow.com/questions/55754906/errors-in-browser-console-requests-to-sockjs-node-infot-1555629946494), that led me to adding 
> ```
>  public: `https://0.0.0.0:${ devPorts ? 8005 : 80 }`,
> ```
> to `shell/vue.config.ts` config.devServer which solved the problem. Can this be added? The change also resolved the issue i was seeing with hot-reload (interestingly it looks like that's done over websocket now, not sure how it worked before).

I'm not sure what's being asked. I believe I already did what you requested.
image

**Review by richard-cox — CHANGES_REQUESTED** [review]
> > image
> > vs
> > image

> TMI But to explain this one... This was caused because the dev script omitted the `source ./scripts/version` prefix. I had verified and tested by setting `DASHBOARD_VERSION` env var which did work. I also found out that yarn uses it's own shell which you can set using `yarn config set script-shell /bin/bash` and I found that out because the default shell didn't have access to `source`.

Interesting. Someone has raised the following PR which might help others in the future
https://github.com/rancher/dashboard/pull/8013

> > > updates to yarn app creator

> > Discussed in slack

> This was a massive pain to figure out and took multiple days and was ultimately a small change. When I first posted this PR this change would've been simple as I had also converted the ui-plugin-examples https://github.com/rancher/ui-plugin-examples/pull/17/files. I really think we need to stop cycling on this PR. Many of the issues raised in this PR could've been resolved in subsequent changes and I think very few would've actually been blockers. At least one of the requests was to account for a change that hadn't even been pushed yet (the versioning).

Dashboard version string and app creator aside, there's been a number of blockers. Some that have only just been resolve or still open include $worker creation, typescript conversions to js and `autoLoadPackages` handling

Either way we need to wait for 2.7.2 to branch before merging. We should be in a good place when it happens.

> > autoLoadPackages isn't wired in (build in /pkg are skipped
> > - Commented above but I don't think this is actually getting used in master. Can you link if you know where it's used?
> 
> I got confused on what the `autoLoad` does. In vue.config.ts
> - `autoLoad` will stick the referenced package from ./pkg/<ref>/<ref>.umd.min.js (unless in the `excludes` value) into the DOM and then `$plugin.loadAsync` from `shell/plugins/plugin.js`
>   - Note - this is wired in via https://github.com/rancher/dashboard/blob/master/shell/nuxt.config.js#L588 , which does not happen in the new world
> - further down packages in `pkg` directories are brought in via another method.
> 
> @nwmac How do these two features work together, is `autoLoad` still needed? Can it be removed or should it be fixed in the new vue world?

Neil might need a prod from us both on this one.

> > that led me to adding 
> > ```
> >  public: `https://0.0.0.0:${ devPorts ? 8005 : 80 }`,
> > ```
> > to `shell/vue.config.ts` config.devServer which solved the problem. Can this be added? The change also resolved the issue i was seeing with hot-reload (interestingly it looks like that's done over websocket now, not sure how it worked before).

> I'm not sure what's being asked. I believe I already did what you requested.
image

Those `port` and `host` entries are in `master`. The current request is for a `public` entry. I had issues before with the `host`, which i now need to manually specify in my host file (at lease machine name --> 0.0.0.0).

File: shell/edit/provisioning.cattle.io.cluster/import.vue
Comment (richard-cox) [reply shell/edit/provisioning.cattle.io.cluster/import.vue:24]: Looks like someone has similar issues when running the shell as a package https://github.com/rancher/dashboard/issues/8008. I'd vote we pull the fix out of this PR and deal with it via #8008

Comment (codyrancher) [reply shell/edit/provisioning.cattle.io.cluster/import.vue]: Fine with me, it's a better fix anyway. I've pulled the latest.

**Comment by codyrancher** [comment]
The public entry was added.

**Comment by codyrancher** [comment]
> Dashboard version string and app creator aside, there's been a number of blockers. Some that have only just been resolve or still open include $worker creation, typescript conversions to js and autoLoadPackages handling

> Either way we need to wait for 2.7.2 to branch before merging. We should be in a good place when it happens.

I debated whether I would respond since I'm not sure it will impact anything but I decided I wanted to just in case it might improve things in the future. I say this with respect, I know you put a good amount of effort and time into these reviews but I disagree with most of the above.

It's so difficult for me to see a world in which the Typescript changes requested were a blocker.
- The changes obviously had 0 impact on the user as it didn't impact correctness at all
- The changes hardly had any impact on a developer since
  - It was only in config files where there's very little logic 
  - Very few places in our code can make use of intellisense or type enforcement by the compiler.
  - Vue 2 has very poor support for typescript and we spend seemingly most our time in the vue files
  - I might be making a bit of a stretch on this one but it sounds like Vue3 has better support for Typescript and Vue3 is blocked by this change. The longer we delay changes like this the longer we delay better support and adoption of Typescript in this project. 

-  You're right that we have to wait for 2.7.2. But this PR has been open for months and now multiple releases... I have to admit that I've been responsible for the delays too but stuff like Typescript which could've easily been recorded as an issue and fixed after this change had been pushed definitely contribute.

**Comment by codyrancher** [comment]
- I've updated the routes and plugins with the latest. 
- I've also updated https://github.com/rancher/ui-plugin-examples/pull/17.

Unless there's something wrong with autoload I don't think there's much left.

**Review by richard-cox — APPROVED** [review]
I've
- rebased to resolve a merge conflict
- fixed websocket proxying for paths other than v1/v3/k8s (for example streaming container logs)
- created https://github.com/rancher/dashboard/issues/8338

There was a failing test, if that passes though and once we branch for 2.7.2, lets get this in.

File: shell/nuxt/App.js
Comment (cnotv) [inline shell/nuxt/App.js:11]: I have some concerns about hardcoding hashed imports, as they may cause cache issues.
Let's make an issue to review this and apply proper naming maybe?

File: shell/nuxt/axios.js
Comment (cnotv) [inline shell/nuxt/axios.js:86]: This, as for the cache, should have an issue to remove Nuxt residual out of the code.

File: shell/nuxt/components/nuxt-build-indicator.vue
Comment (cnotv) [inline shell/nuxt/components/nuxt-build-indicator.vue:1]: Can we document/comment concisely somewhere what is this code from the `shell/nuxt` folder for, why do we need it and what are we planning to do?
Up to you either with a commit or an issue.

File: shell/tsconfig.default.json
Comment (cnotv) [inline shell/tsconfig.default.json:17]: It does not make so much sense to have a default inside a folder and the one in the root importing it which imports another one 😅 
We should make an issue where we review the TsConfigs we have around.

**Review by cnotv — COMMENTED** [review]
I'm just doing an overview of the changes and I think there's a bunch of work which needs to be addressed and added in the TODO for the Vue3 migration. 
It makes no sense to block for this, obviously, but we need to take notes.
https://github.com/rancher/dashboard/issues/7653

ASAP I'll try to run and build it as part of the QA.
Some concerns go to the deployment process and the configuration with the paths, but it seems formally ok.

**Review by richard-cox — APPROVED** [review]
Given https://github.com/rancher/dashboard/issues/8376 is addressed soon, and we're not publishing shell, lets get this in.

**Pull request merged**
